#!/usr/bin/env python

PACKAGE='chapter3_tutorials'

from math import pi
from dynamic_reconfigure.parameter_generator_catkin import *
from driver_base.msg import SensorLevels

gen = ParameterGenerator()

gen.add('bool_param', bool_t, SensorLevels.RECONFIGURE_RUNNING,
        'Bool param', True)
gen.add('int_param', int_t, SensorLevels.RECONFIGURE_STOP,
        'Int param', 0, -10, 10)
gen.add('double_param', double_t, SensorLevels.RECONFIGURE_CLOSE,
        'Double param', 0.0, -pi, pi)

foo = gen.const('foo', str_t, 'Foo', 'FOO')
bar = gen.const('bar', str_t, 'Bar', 'BAR')
baz = gen.const('baz', str_t, 'Baz', 'BAZ')
strings = gen.enum([foo, bar, baz], 'Strings')
gen.add('string_param', str_t, SensorLevels.RECONFIGURE_RUNNING,
        'String param', 'Foo', edit_method = strings)

exit(gen.generate(PACKAGE, PACKAGE, 'DynamicParam'))
